### DISCUSSION TITLE
Component versioning and instantiation of previous versions

### DISCUSSION THREAD

**Discussion by jan-david-fischbach** [gdsfactory/gdsfactory#708] [Q&A] [answered]
Hello Everyone,
what is your strategies to version components and retrieve specific versions at will? Sometimes it might become necessary to introduce breaking changes to components. It would be great if there was a way to avoid breaking legacy devices doing so. Is there any reasonable way to tag specific versions of a component and use the specified version regardless of future updates similar to specifying a version in your favorite package manager? As far as I can see this would require the code for all tagged versions to be present in the HEAD, which feels like a git nightmare.  Any Ideas?
Regards 
JD

**Comment by joamatab** [comment, marked as answer]
Hi Jan-David,

Each `Component` has a version and a changelog,

if you want to keep using two components you can use a new pcell function for that

if you just want to update a Pcell and want to track its changes you can add them to the changelog

**Reply by jan-david-fischbach** [reply]
Hey Joaquin,
i have seen the version number in the metadata. I however feel a bit lost on how to use it properly. Could you maybe point me to the corresponding documentation? I seem to not be able to find it, sorry :/

**Reply by joamatab** [reply]
https://gdsfactory.github.io/gdsfactory/notebooks/08_pdk.html#Version-control-components

**Reply by jan-david-fischbach** [reply]
Thanks I seem to have somehow overlooked it... Thanks 👍

**Reply by joamatab** [reply]
I added it yesterday:) let me know further improvements
